### PR TITLE
stability tests: add s3 stability test template and suite

### PR DESCRIPTION
**Pull request by matthewfala**
Add Test suite for S3 stability tests and also sensitivity analysis.

There are 40 low throughput and 40 high throughput s3 tests with generic config.
Then there are 5 tests each of several configuration parameter augmentations.

### CODE REVIEW

File: apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/case-config.json
Comment (PettitWesley) [inline apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/case-config.json:1]: why sensitivity?

Comment (PettitWesley) [reply apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/case-config.json:1]: what that word

File: apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-stability/cases/s3-throughput-30mbps.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "definitions": {
+        "datajet_kbps_throughput_tcp": "30000"
```
Comment (PettitWesley) [inline apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-stability/cases/s3-throughput-30mbps.json:3]: future: split throughput through multiple streams of input, more like golden path

File: apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/case-config.json
Comment (PettitWesley) [reply apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/case-config.json:1]: these are extreme cases?

File: apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-stability/cases/s3-throughput-30mbps.json (hunk 1 above)
Comment (PettitWesley) [reply apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-stability/cases/s3-throughput-30mbps.json:3]: real world high throughput use cases are probably k8s where throughput is spread over multiple tags from many pods

File: apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-stability/cases/s3-throughput-30mbps.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+    }
+}
```
Comment (matthewfala) [inline apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-stability/cases/s3-throughput-30mbps.json:5]: Possibly best to spread the output among multiple s3 outputs.
- Use multiple inputs or multiple tags
   - Tcp -> tcp1 -> 10mb/s
   - Tcp -> tcp2 -> 10mb/s
   - Tcp -> tcp3 -> 10mb/s

- One S3 Output

File: apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/cases/s3-throughput-100mbps.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+    "definitions": {
+        "s3_use_put_object": "Off",
```
Comment (matthewfala) [inline apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/cases/s3-throughput-100mbps.json:3]: Get rid of this. It is the default.

File: apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/cases/s3-throughput-100mbps.json
Comment (PettitWesley) [inline apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/cases/s3-throughput-100mbps.json:1]: 100 MBPS vs file says KBPS!!

File: apps/firelens-stability/templates/s3-fargate-v04-05-2023/fluent-bit-onepod.conf
Comment (PettitWesley) [inline apps/firelens-stability/templates/s3-fargate-v04-05-2023/fluent-bit-onepod.conf:1]: add a comment on why this file is empty to make the file non-empty

File: apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/case-config.json
Comment (PettitWesley) [reply apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/case-config.json:1]: if the goal is to test a bunch of different parameters, there are a lot that these tests don't set: https://docs.fluentbit.io/manual/pipeline/outputs/s3/

Comment (PettitWesley) [reply apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/case-config.json:1]: s3_key_format, compression, feel like major ones we should test

File: apps/firelens-stability/templates/s3-fargate-v04-05-2023/fluent-bit.conf [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    # Mem_Buf_Limit <= Memory_limit / 2 / (number of INPUT sections + 1)
+    # 64MB = 512MB / 2 / (4 INPUT sections)
+    Mem_Buf_Limit 64MB
+
+{{#times definitions.s3_outputPluginCount}}
+[OUTPUT]
+    Name s3
+    Match ApplicationLogs
+    bucket {{definitions.s3LogDestinationBucketName}}
+    s3_key_format /{{managed.executionName}}-{{managed.executionId}}/{{managed.caseNameUnique}}/${RUN_ID}/%Y-%m-%d/$UUID
```
Comment (matthewfala) [inline apps/firelens-stability/templates/s3-fargate-v04-05-2023/fluent-bit.conf:44]: Create test with all weird things
- Tag of n
- $index
- UUID

Anything else

File: apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/case-config.json
Comment (PettitWesley) [reply apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/case-config.json:1]: one with all special s3_key_format features

File: apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/cases/s3-outputPluginCount-50.json [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{
+    "definitions": {
+        "s3_outputPluginCount": "50"
```
Comment (matthewfala) [inline apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/cases/s3-outputPluginCount-50.json:3]: Reduce to 10 plugins

Comment (PettitWesley) [inline apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/cases/s3-outputPluginCount-50.json:3]: how about 10

File: apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/cases/s3-throughput-100mbps.json
Comment (matthewfala) [reply apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/cases/s3-throughput-100mbps.json:1]: resolved.

File: apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/case-config.json
Comment (matthewfala) [reply apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/case-config.json:1]: Added a special s3_key_format case

File: apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/cases/s3-throughput-100mbps.json
Comment (matthewfala) [reply apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/cases/s3-throughput-100mbps.json:1]: Great catch!

File: apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/cases/s3-throughput-100mbps.json [hunk 6]
```diff
@@ -0,0 +1,5 @@
+{
+    "definitions": {
+        "datajet_kbps_throughput_four_tcp": 25000
```
Comment (matthewfala) [inline apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/cases/s3-throughput-100mbps.json:3]: This is sent to 4 different tcp streams, all to one s3 plugin.

File: apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-stability/cases/s3-throughput-30mbps.json [hunk 7]
```diff
@@ -0,0 +1,5 @@
+{
+    "definitions": {
+        "datajet_kbps_throughput_four_tcp": "7500"
```
Comment (matthewfala) [inline apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-stability/cases/s3-throughput-30mbps.json:3]: This is the amount of data sent to fluent bit. The actual data stored in s3 may be more as it includes ECS metadata like the following:
```
{"date":"2023-04-12T19:20:59.298693Z","log":"{\"log\":\"<1kb of data here>\"}","ecs_cluster":"DEMO-2_28_5-stability-tests-r3-E-2023-04-12T191930525-iTZ4MHNx-ecs-firelens-stability-tests","ecs_task_arn":"arn:aws:ecs:us-west-2:210752607241:task/DEMO-2_28_5-stability-tests-r3-E-2023-04-12T191930525-iTZ4MHNx-ecs-firelens-stability-tests/79b9e0854bbb4917a7a466d1aaea7f23","ecs_task_definition":"ecs-firelens-stability-tests-s3-sensitivity-s3-s3_compression-gzip:10"}
```

File: apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/cases/s3-throughput-100mbps.json
Comment (PettitWesley) [reply apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/cases/s3-throughput-100mbps.json:1]: the file I see still says MPBS??

Comment (matthewfala) [reply apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/cases/s3-throughput-100mbps.json:1]: I increased it to 100000kbps, so 100mbps

File: apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-stability/cases/s3-throughput-30mbps.json (hunk 1 above)
Comment (matthewfala) [reply apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-stability/cases/s3-throughput-30mbps.json]: Split it to 4 different tcps

File: apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-stability/cases/s3-throughput-30mbps.json (hunk 2 above)
Comment (matthewfala) [reply apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-stability/cases/s3-throughput-30mbps.json]: This is done.

File: apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/cases/s3-throughput-100mbps.json (hunk 3 above)
Comment (matthewfala) [reply apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/cases/s3-throughput-100mbps.json]: This is done.

File: apps/firelens-stability/templates/s3-fargate-v04-05-2023/fluent-bit-onepod.conf
Comment (matthewfala) [reply apps/firelens-stability/templates/s3-fargate-v04-05-2023/fluent-bit-onepod.conf:1]: Discussed and noted that empty files are allowed by git

File: apps/firelens-stability/templates/s3-fargate-v04-05-2023/fluent-bit.conf (hunk 4 above)
Comment (matthewfala) [reply apps/firelens-stability/templates/s3-fargate-v04-05-2023/fluent-bit.conf]: Added a test case with the following

File: apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/cases/s3-outputPluginCount-50.json (hunk 5 above)
Comment (matthewfala) [reply apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/cases/s3-outputPluginCount-50.json]: Done

**Review by PettitWesley — APPROVED** [review]
I am concerned about the data size and likelihood of OOMKill in the 100MBPS test

File: apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/cases/s3-throughput-10mbps.json [hunk 8]
```diff
@@ -0,0 +1,5 @@
+{
+    "definitions": {
+        "datajet_kbps_throughput_four_tcp": 2500
```
Comment (matthewfala) [inline apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-sensitivity/cases/s3-throughput-10mbps.json:3]: I updated the test case to 10mbps, 2.5mbps to 4 tcp inputs.

**Comment by matthewfala** [comment]
Merged to the stability wip branch!

**Pull request merged**